File: opgg.py
from flask import Flask,render_template,request
import os
import requests
from pprint import pprint as pp

app = Flask(__name__, template_folder="views")
@app.route('/')
def index():
    return render_template('index1.html')
    
@app.route('/search')
def search():
    sum_name = request.args.get('name')
    url = "https://kr.api.riotgames.com/lol/summoner/v4/summoners/by-name/{}".format(sum_name)
    headers = {
        "Origin": "https://developer.riotgames.com",
        "Accept-Charset": "application/x-www-form-urlencoded; charset=UTF-8",
        "X-Riot-Token": "RGAPI-53dcf13b-6c33-420b-bdbc-7e97ade70488",
        "Accept-Language": "ko-KR,ko;q=0.9,en-US;q=0.8,en;q=0.7",
        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Safari/537.36"
    }
    res = requests.get(url=url,headers=headers)
    encrypted_id = res.json()['id']
    url_league = "https://kr.api.riotgames.com/lol/league/v4/positions/by-summoner/{}".format(encrypted_id)
    res_league = requests.get(url=url_league,headers=headers)
    league_dicts = res_league.json()
    #pp(league_dicts)
    #print('-------------------------------')
    def get_league_info(league_dict):
        res = [
        league_dict.get('queueType'),
        league_dict.get('tier'),
        league_dict.get('rank'),
        league_dict.get('wins'),
        league_dict.get('losses'),
        league_dict.get('leagueName'),
        league_dict.get('leaguePoints')
            ]
        return res
    results = []
    for league_dict in league_dicts:
        results.append(get_league_info(league_dict))
    length = len(results)
    
    return render_template('search.html',sum_name=sum_name,results=results,length=length)

File: opgg.php
<?
	$text = $_GET['summoner_name'];
		
	$summoner_name = urlencode($text);
	$api_key="RGAPI-53dcf13b-6c33-420b-bdbc-7e97ade70488i";
	$url = "https://kr.api.riotgames.com/lol/summoner/v3/summoners/by_name/".$summoner_name."?api_key=".$api_key;

	$is_post = false;
	$ch = curl_init();

	curl_setopt($ch, CURLOPT_URL, $url);
	curl_setopt($ch, CURLOPT_URL, $is_post);
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
	curl_close($ch);

	$result = json_decode($response, true);
